ModPatches/AOC The Cleanup Devil/Defs/AOC The Cleanup Devil/Ammo_PowerCell.xml

```diff
@@ -1,95 +1,345 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Defs>
 
-	<!-- AOCPOWERCELL -->
-
 	<ThingCategoryDef>
 		<defName>AmmoAOCPowerCell</defName>
-		<label>AOCPOWERCELL</label>
-		<parent>AmmoRifles</parent>
-		<iconPath>UI/Icons/ThingCategories/CaliberRifle</iconPath>
+		<label>AOC POWERCELL</label>
+		<parent>AmmoAdvanced</parent>
+		<iconPath>UI/Icons/ThingCategories/CaliberCharge</iconPath>
 	</ThingCategoryDef>
 
-	<!-- ==================== Ammo AOCPOWERCELL ========================== -->
+	<!-- ==================== AmmoSet ========================== -->
+
+	<CombatExtended.AmmoSetDef>
+		<defName>AmmoSet_AOCPowerCell</defName>
+		<label>AOC POWER CELL</label>
+		<ammoTypes>
+			<Ammo_AOCPowerCell_HEAT>Bullet_AOCPowerCell_HEAT</Ammo_AOCPowerCell_HEAT>
+			<Ammo_AOCPowerCell_Thermobaric>Bullet_AOCPowerCell_Thermobaric</Ammo_AOCPowerCell_Thermobaric>
+			<Ammo_AOCPowerCell_EMP>Bullet_AOCPowerCell_EMP</Ammo_AOCPowerCell_EMP>
+		</ammoTypes>
+	</CombatExtended.AmmoSetDef>
+
+	<!-- ==================== Ammo ========================== -->
 
-	<ThingDef Class="CombatExtended.AmmoDef" Name="AOCPowerCellBase" ParentName="SmallAmmoBase" Abstract="True">
-		<description>A power cell.</description>
+	<ThingDef Class="CombatExtended.AmmoDef" Name="AOCPowerCellBase" ParentName="SpacerMediumAmmoBase" Abstract="True">
+		<description>Power cells used as ammunition for the AOC Suvi Buster.</description>
 		<statBases>
-			<Mass>0.5</Mass>
-			<Bulk>0.1</Bulk>
+			<MaxHitPoints>100</MaxHitPoints>
 		</statBases>
 		<tradeTags>
 			<li>CE_AutoEnableTrade</li>
-			<li>CE_AutoEnableCrafting</li>
+			<li MayRequire="Ludeon.RimWorld.Biotech">CE_AutoEnableCrafting_FabricationBench</li>
 		</tradeTags>
 		<thingCategories>
 			<li>AmmoAOCPowerCell</li>
 		</thingCategories>
+		<stackLimit>40</stackLimit>
+		<cookOffFlashScale>40</cookOffFlashScale>
+		<cookOffSound>MortarBomb_Explode</cookOffSound>
 	</ThingDef>
 
 	<ThingDef Class="CombatExtended.AmmoDef" ParentName="AOCPowerCellBase">
-		<defName>Ammo_AOCPowerCell</defName>
-		<label>(AOC) Power cell</label>
+		<defName>Ammo_AOCPowerCell_HEAT</defName>
+		<label>AOC PowerCell (HEAT)</label>
 		<graphicData>
-			<texPath>Things/Ammo/Charged/MediumMech</texPath>
+			<texPath>Things/Ammo/AOCPowerCell/HEAT</texPath>
 			<graphicClass>Graphic_StackCount</graphicClass>
 		</graphicData>
-		<ammoClass>Charged</ammoClass>
+		<statBases>
+			<Mass>0.5</Mass>
```

Comment:
Since these stats are identical across all three variants of the ammo, it makes more sense to keep them in the base def.